### DISCUSSION TITLE
Delete expired token from db

### DISCUSSION THREAD

**Discussion by diegocgaona** [BeanieODM/beanie#393] [Question] [answered]
I have a collection with some tokens and I want to delete then after they are expired. Example:

```
  {
    _id: ObjectId("6XXX2f82c843ef1dae16fe55"),
    token: 'XXXXLMCkcTLH8aUOnMb83ob6u4v0cgKmMxuk0IU',
    user_id: ObjectId("63XX81c02bea72a94db8397f5"),
    created_at: ISODate("2022-10-25T18:48:34.457Z")
  }
```

After each login I want to delete all the old tokens, based on a time in second (example, lifetime=3600).

I will need to compare the date on my item (field created_at) with my lifetime parameter.
How I can do that with Beanie?

Would something like this work? How to compare datas for delete on Beanie?
And if someone could help, how to calc the date time (from now - 3600 seconds)?

`await Reports.find(tokens.created_at < "2022-10-25T18:00:34.457Z").delete()`

Thanks!!

**Comment by roman-right** [comment, marked as answer]
Hey! I think TTL indexes will work in your case. You can add it using `IndexModel`: https://beanie-odm.dev/tutorial/indexes/#multi-field-indexes . Here is the MongoDB doc about TTL indexes: https://www.mongodb.com/docs/manual/core/index-ttl/
